install_config/redeploying_certificates.adoc

```diff
@@ -11,63 +11,202 @@
+. Run the *_redeploy-certificates.yml_* playbook with the
+`openshift_certificates_redeploy_ca=true` variable, specifying your inventory
+file:
++
+----
+$ ansible-playbook -i <inventory_file> \
+    playbooks/byo/openshift-cluster/redeploy-certificates.yml \
+    --extra-vars "openshift_certificates_redeploy_ca=true"
+----
+
+. As discussed previously, the playbook now serially evacuates all nodes in the
+cluster, requiring manual redeployment of registry and router certificates. See
+xref:redeploying-registry-and-router-certificates[Redeploying Registry and
+Router Certificates] to complete this procedure.
+
+[[redeploying-registry-and-router-certificates]]
+== Registry and Router Certificates
+
+When nodes are evacuated due to a redeployed CA, registry and router pods are
+restarted. This can cause outages because they cannot reach the masters using
+their old certificates.
+
+To address this issue, or to redeploy these certificates for any reason, you can
+manually redeploy registry and router certificates.
+
+[[redeploying-registry-certificates-manually]]
+=== Redeploying Registry Certificates Manually
+
+To redeploy registry certificates manually, you must add new registry
+certificates to a secret named `registry-certificates`, then redeploy the
+registry:
+
+. Set the following environment variables locally to make later commands less
+complex:
++
+----
+$ REGISTRY_IP=`oc get -n default service docker-registry -o jsonpath='{.spec.clusterIP}'`
+$ REGISTRY_HOSTNAME=`oc get -n default route/docker-registry -o jsonpath='{.spec.host}'`
+----
+
+. Create new registry certificates:
++
+----
+$ oc adm ca create-server-cert \
+  --signer-cert=/etc/origin/master/ca.crt \
+  --signer-key=/etc/origin/master/ca.key \
+  --hostnames=$REGISTRY_IP,docker-registry.default.svc.cluster.local,$REGISTRY_HOSTNAME \
+  --cert=/etc/origin/master/registry.crt \
+  --key=/etc/origin/master/registry.key \
+  --signer-serial=/etc/origin/master/ca.serial.txt
+----
+
+. Update the `registry-certificates` secret with the new registry certificates:
++
+----
+$ oc secret new registry-certificates \
+  /etc/origin/master/registry.crt \
+  /etc/origin/master/registry.key \
+  -n default \
+  -o json | oc replace -f -
+----
+
+. Redeploy the registry:
++
+----
+$ oc deploy -n default dc/docker-registry --latest
+----
+
+[[redeploying-router-certificates-manually]]
+=== Redeploying Router Certificates Manually
+
+The router is secured using a
+xref:../dev_guide/secrets.adoc#service-serving-certificate-secrets[service serving certificate secret], which was automatically created when an annotation
+was adding to the router service during initial deployment.
+
+To redeploy router certificates manually, that service serving certificate can
+be triggered to be recreated by deleting the secret, clearing and re-adding
+annotations to the `router` service, then redeploying the router:
+
+. Delete the `router-certs` secret:
++
+----
+$ oc delete -n default secret router-certs
```

Comment:
@abutcher I didn't have this secret but I had a router deployed. And I didn't have a hit on the `router-certs` name across openshift-docs. Is this maybe something from a newer-ish version of the installer? My particular cluster was initially installed as 3.1.1.